griptape/tools/email_client/tool.py

```diff
@@ -1,56 +1,70 @@
-        imap_url: The hostname or url of the SMTP server (imap.gmail.com).
-        imap_user: Setting this will override whatever is set as the username parameter for IMAP activities.
-        imap_password: Setting this will override whatever is set as the username parameter for IMAP activities.
-    """
+    """Tool for working with email"""
+
     # if you set imap|smtp creds explicitly these fields will be overridden
     username: Optional[str] = field(default=None, kw_only=True)
+    """Used by default for both SMTP and IMAP activities. Will be used as the from email when sending emails."""
     password: Optional[str] = field(default=None, kw_only=True)
+    """Used by default for both SMTP and IMAP activities."""
 
     smtp_host: Optional[str] = field(default=None, kw_only=True)
+    """The hostname or url of the SMTP server (smtp.gmail.com)."""
     smtp_port: Optional[int] = field(default=None, kw_only=True)
+    """The port name of the SMTP server (465)."""
     smtp_use_ssl: bool = field(default=True, kw_only=True)
+    """Should EmailClient use SSL when sending."""
 
     # if you set the smtp user/password fields they will override
-    smtp_user: Optional[str] = field(default=None, kw_only=True)
-    smtp_password: Optional[str] = field(default=None, kw_only=True)
+    smtp_user: Optional[str] = field(default=Factory(lambda self: self.username, takes_self=True), kw_only=True)
+    """Setting this will override whatever is set as the username parameter for SMTP activities."""
+    smtp_password: Optional[str] = field(default=Factory(lambda self: self.password, takes_self=True), kw_only=True)
+    """Setting this will override whatever is set as the username parameter for SMTP activities."""
 
     imap_url: Optional[str] = field(default=None, kw_only=True)
+    """The hostname or url of the IMAP server (imap.gmail.com)."""
 
     # if you set imap user/password fields they will override
-    imap_user: Optional[str] = field(default=None, kw_only=True)
-    imap_password: Optional[str] = field(default=None, kw_only=True)
+    imap_user: Optional[str] = field(default=Factory(lambda self: self.username, takes_self=True), kw_only=True)
+    """Setting this will override whatever is set as the username parameter for IMAP activities."""
+    imap_password: Optional[str] = field(default=Factory(lambda self: self.password, takes_self=True), kw_only=True)
+    """Setting this will override whatever is set as the username parameter for IMAP activities."""
 
     email_max_retrieve_count: Optional[int] = field(default=None, kw_only=True)
+    """Used to limit the number of messages retrieved during any given activities."""
 
+    # TODO: Ask in PR if mailboxes is unused. Does the mention in the @activity decorator
+    #       provide some functionality or was it was aspirational?
     mailboxes: Optional[dict[str, str]] = field(default=None, kw_only=True)
```

Comment:
The description will be rendered in this [jinja template](https://github.com/griptape-ai/griptape/blob/main/griptape/templates/tasks/partials/_tool.j2#L5) that is passed to the LLM. So yeah, `mailboxes` is still used.

Reply:
- Documented expected usage in the class docstring including an example.
- Added a unit test to verify the mailbox descriptions are rendered in the LLM input. (The unit test was initially just to check my own understanding, but it seems useful, so I included it in the commit).